core/keyserver/keyserver.go

```diff
@@ -363,3 +367,57 @@ func (s *Server) GetDomain(ctx context.Context, in *pb.GetDomainRequest) (*pb.Do
 		Vrf:      domain.VRF,
 	}, nil
 }
+
+func (s *Server) latestLogRoot(ctx context.Context, d *domain.Domain) (*tpb.SignedLogRoot, error) {
+	// Fresh Root.
+	logRoot, err := s.tlog.GetLatestSignedLogRoot(ctx,
+		&tpb.GetLatestSignedLogRootRequest{
+			LogId: d.LogID,
+		})
+	if err != nil {
+		glog.Errorf("tlog.GetLatestSignedLogRoot(%v): %v", d.LogID, err)
+		return nil, status.Errorf(codes.Internal, "Cannot fetch SignedLogRoot")
+	}
+	sth := logRoot.GetSignedLogRoot()
+	return sth, nil
+}
+
+// latestLogRootProof returns the lastest SignedLogRoot and it's consistency proof.
+func (s *Server) latestLogRootProof(ctx context.Context, d *domain.Domain, firstTreeSize int64) (*tpb.SignedLogRoot, *tpb.Proof, error) {
+
+	sth, err := s.latestLogRoot(ctx, d)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Consistency proof.
+	secondTreeSize := sth.GetTreeSize()
+	var logConsistency *tpb.GetConsistencyProofResponse
+	if firstTreeSize != 0 {
+		logConsistency, err = s.tlog.GetConsistencyProof(ctx,
+			&tpb.GetConsistencyProofRequest{
+				LogId:          d.LogID,
+				FirstTreeSize:  firstTreeSize,
+				SecondTreeSize: secondTreeSize,
+			})
+		if err != nil {
+			glog.Errorf("tlog.GetConsistency(%v, %v, %v): %v",
+				d.LogID, firstTreeSize, secondTreeSize, err)
+			return nil, nil, status.Errorf(codes.Internal, "Cannot fetch log consistency proof")
+		}
+	}
+	return sth, logConsistency.GetProof(), nil
+}
+
+// latestRevision returns the latest map revision, given the latest sth.
+// The log is the authoritative source of the latest revision.
```

Comment:
Naming seems a bit odd given it takes a log STH and returns a map revision. mapRevisionFor() maybe?

Reply:
cool